lib/shred.ts

```diff
@@ -25,11 +25,29 @@ const parquet_schema = require('./schema');
  *   }
  *
  */
-exports.shredRecord = function(schema, record, buffer) {
+
+// Replace with imported Parquet Schema when merged with Typescript version of schema
+interface ParquetSchema {
+  schema: SchemaDefinition
+  fields: Record<string, ParquetField>
+  fieldList: Array<ParquetField>
+}
+
+interface RecordBuffer {
+  columnData: Record<string, ParquetData>
+  rowCount: number,
+  pageRowCount: number,
+  pages: Record<string,object>
+}
+
+type RecordTypes = string | number | boolean
+
+export const shredRecord = function(schema: ParquetSchema, record: Record<string, RecordTypes | RecordTypes[]>, buffer: RecordBuffer) {
```

Comment:
Record types is just what kind of values that a user can pass in after defining a schema. In the shredRecord function, the record parameter is not exactly correct right now because it does not take in account nested fields